ui/app/rif/pages/domainsPage.js

```diff
@@ -1,49 +1,104 @@
-import { Component } from 'react';
-import h from 'react-hyperscript'
+import React, { Component } from 'react';
+import PropTypes from 'prop-types'
 import { connect } from 'react-redux'
+import Select from "react-dropdown-select"
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faChevronLeft } from '@fortawesome/free-solid-svg-icons'
 import actions from '../../actions'
+import rifActions from '../../rif/actions'
+
+function statusStyle(status){
+	switch(status){
+		case 'active':
+			return 'chiplet-status-active'
+		case 'pending':
+			return 'chiplet-status-pending'
+		case 'expired':
+			return 'chiplet-status-expired'
+		case 'expiring':
+			return 'chiplet-status-expiring'
+	}
+}
 
 class DomainsScreen extends Component {
-  navigateTo (url) {
-    global.platform.openWindow({ url })
-  }
+	state = {
+		domains: [],
+	}
+	constructor(props) {
+		super(props);
+		if(localStorage.rnsDomains){
+			let domains = JSON.parse(localStorage.rnsDomains);
+			this.state = { 
+				domains: domains,
+			}
+		}	
+	}
+	navigateTo (url) {
+		global.platform.openWindow({ url })
+	}
+
+	chiplet = (data, id) => {
+		return <div id="chiplet" className={'chiplet'} key={id}>
+			<div className={'chiplet-body'}>
+				<div onClick={() => {this.props.showDomainsDetailPage(data)}} id="chipletTitle" className={'chiplet-title'}>
+					{data.domain}
+				</div>
+				<div id="chipletDescription" className={'chiplet-description'}>	
+					<div id="chipletExpiration">
+						<span>Expires on: {data.expiration}</span>
+					</div>
+					<div id="chipletRenew">
+						<span>Auto-renew: {data.autoRenew ? 'on' : 'off'}</span>
+					</div>
+				</div>
+			</div>
+			<div className={'chiplet-status-wrapper ' + statusStyle(data.status)}>
+				<div id="chipletStatus" className={'chiplet-status-text'}>
+					{data.status}
+				</div>
+			</div>
+		</div>
+	}
+
   render () {
-    return (
-      h('.flex-column.flex-grow', {
-        style: {
-          maxWidth: '400px',
-        },
-      }, [
-        // subtitle and nav
-        h('.section-title.flex-row.flex-center', [
-          h('i.fa.fa-arrow-left.fa-lg.cursor-pointer', {
-            onClick: (event) => {
-              this.props.dispatch(actions.goHome())
-            },
-            style: {
-              position: 'absolute',
-              left: '30px',
-            },
-          }),
-          h('h2', 'Domains Screen'),
-        ]),
-        // main view
-        h('.flex-column.flex-justify-center.flex-grow.select-none', [
-          h('.flex-space-around', {
-            style: {
-              padding: '0 30px',
-            },
-          }, [
-            h('.info', [
-              h('div', 'RNS Domains'),
-            ]),
-          ]),
-        ]),
-      ])
-    )
+	return (
+	  <div className={'body'}>
+		<FontAwesomeIcon icon={faChevronLeft} className={'rif-back-button'} onClick={() => this.props.goHome()}/>
+		<Select 
```

Comment:
Why are we using a Select instead of a normal search bar?

Reply:
Its the name of the component. 
https://www.npmjs.com/package/react-dropdown-select